pyodide-build/pyodide_build/buildall.py

```diff
@@ -477,6 +478,48 @@ def _generate_package_hash(full_path: Path) -> str:
     return sha256_hash.hexdigest()
 
 
+def parse_top_level_import_name(whlfile: Path) -> list[str] | None:
+    """
+    Parse the top-level import name from a package file.
+    """
+
+    assert whlfile.name.endswith(".whl"), "Not a package file"
+
+    pkg_name_ver = "-".join(whlfile.name.split("-")[:2])
+    dist_info_dir = PurePath(f"{pkg_name_ver}.dist-info")
+    top_level_filepath = str(dist_info_dir / "top_level.txt")
+
+    whlzip = zipfile.Path(whlfile)
+    top_level_file = whlzip / top_level_filepath
+    if top_level_file.is_file():
+        # only packages built with setuptools has `top_level.txt` file.
+        # So if the package is built with flit or poetry, it doesn't exist.
+        # Note that `top_level.txt` file comes from deprecated egg format,
+        # so even setuptools may remove this someday.
+        return top_level_file.read_text().strip().split("\n")
+
+    # If there is no top_level.txt file, we will find top level imports by
+    # 1) a python file on a top-level directory
+    # 2) a sub directory with __init__.py
+    # following: https://github.com/pypa/setuptools/blob/d680efc8b4cd9aa388d07d3e298b870d26e9e04b/setuptools/discovery.py#L122
+    top_level_imports = []
+    for subdir in whlzip.iterdir():
+        if subdir.is_file() and subdir.name.endswith(".py"):
```

Comment:
We should also skip if the subdir if it cannot be an import name (e.g. contains `-` or spaces) I think.

Otherwise currently it finds a folder with `-` as an import name.